### PR TITLE
feat: add kbd67v2 support

### PR DESCRIPTION
**Pull request by Vorror**
## Description

Add support for the second revision of the KBD67 pcb.

## Types of Changes

- [ ] Core
- [ ] Bugfix
- [ ] New feature
- [ ] Enhancement/optimization
- [x] Keyboard (addition or update)
- [ ] Keymap/layout/userspace (addition or update)
- [ ] Documentation

## Checklist

- [x] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [x] I have read the [**CONTRIBUTING** document](https://docs.qmk.fm/#/contributing).
- [ ] I have added tests to cover my changes.
- [ ] I have tested the changes and verified that they work and don't break anything (as well as I can manage).

### CODE REVIEW

**Comment by noroadsleft** [comment]
Posting to have it publicly documented:

It appears that the rev2 supports all the same layouts that the rev1 does, but has a totally different pin configuration.

That being the case, I'm not 100% keen about copying the two keymaps that are here – they're exact copies of the `default` and `koba` keymaps for rev1 – but I haven't worked out if there's a good solution for that yet, considering keymaps for the hotswap version PCB aren't 100% compatible with either of the other revisions.

**Comment by mechmerlin** [comment]
I would model how this works using https://github.com/qmk/qmk_firmware/tree/master/keyboards/clueboard/66

While each rev is different, they all still share the same keymap directory. default and koba keymaps should be made to work for both.

**Comment by noroadsleft** [comment]
@mechmerlin I think we might have to refactor the KBD67 Hotswap for that to work fully; specifically by integrating its layout macro into the common KBD67 codebase.

**Comment by mechmerlin** [comment]
> @mechmerlin I think we might have to refactor the KBD67 Hotswap for that to work fully; specifically by integrating its layout macro into the common KBD67 codebase.

Oof. Ok perhaps do that in another PR then.

File: keyboards/kbdfans/kbd67/rev2/rules.mk [hunk 1]
```diff
@@ -0,0 +1,80 @@
...
+# if this doesn't work, see here: https://github.com/tmk/tmk_keyboard/wiki/FAQ#nkro-doesnt-work
+NKRO_ENABLE = no            # USB Nkey Rollover
+BACKLIGHT_ENABLE = yes       # Enable keyboard backlight functionality on B7 by default
+RGBLIGHT_ENABLE = yes        # Enable keyboard RGB underglow
+MIDI_ENABLE = no            # MIDI support (+2400 to 4200, depending on config)
+UNICODE_ENABLE = no         # Unicode
+BLUETOOTH_ENABLE = no       # Enable Bluetooth with the Adafruit EZ-Key HID
+AUDIO_ENABLE = no           # Audio output on port C6
+FAUXCLICKY_ENABLE = no      # Use buzzer to emulate clicky switches
+HD44780_ENABLE = no 		# Enable support for HD44780 based LCDs (+400)
```
Comment (mechmerlin) [inline keyboards/kbdfans/kbd67/rev2/rules.mk]: We should add this line so people with community layouts using `65_ansi` can make use of it. 

```suggestion
HD44780_ENABLE = no 		# Enable support for HD44780 based LCDs (+400)

LAYOUTS = 65_ansi
```

Comment (drashna) [reply keyboards/kbdfans/kbd67/rev2/rules.mk]: @Vorror Any update on this?

File: keyboards/kbdfans/kbd67/rev2/keymaps/default/config.h [hunk 2]
```diff
@@ -0,0 +1,19 @@
+/* Copyright 2018 MechMerlin
```
Comment (mechmerlin) [inline keyboards/kbdfans/kbd67/rev2/keymaps/default/config.h]: I don't remember writing these files =P

File: keyboards/kbdfans/kbd67/rev2/keymaps/default/keymap.c [hunk 3]
```diff
@@ -0,0 +1,99 @@
+/* Copyright 2018 'mechmerlin'
```
Comment (mechmerlin) [inline keyboards/kbdfans/kbd67/rev2/keymaps/default/keymap.c]: You should probably change this along with the year.

File: keyboards/kbdfans/kbd67/rev2/readme.md [hunk 4]
```diff
@@ -0,0 +1,13 @@
+# kbd67
```
Comment (mechmerlin) [inline keyboards/kbdfans/kbd67/rev2/readme.md]:
```suggestion
# KBD67 rev2
```

File: keyboards/kbdfans/kbd67/rev2/rev2.h [hunk 5]
```diff
@@ -0,0 +1,57 @@
+/* Copyright 2018 MechMerlin
```
Comment (mechmerlin) [inline keyboards/kbdfans/kbd67/rev2/rev2.h]: You should probably change this along with the date.

File: keyboards/kbdfans/kbd67/rev2/keymaps/default/config.h (hunk 2 above)
Comment (noroadsleft) [reply keyboards/kbdfans/kbd67/rev2/keymaps/default/config.h]: @mechmerlin These are all copied from the `rev1` files, so you actually *did* write them.

Comment (Vorror) [reply keyboards/kbdfans/kbd67/rev2/keymaps/default/config.h]: @mechmerlin @noroadsleft Correct, which is why I left the original author.

**Pull request merged**